contracts/contracts/vault/VaultCore.sol

```diff
@@ -253,8 +255,42 @@ contract VaultCore is VaultStorage {
                 IStrategy strategy = IStrategy(depositStrategyAddr);
                 // Transfer asset to Strategy and call deposit method to
                 // mint or take required action
-                asset.safeTransfer(address(strategy), allocateAmount);
-                strategy.deposit(address(asset), allocateAmount);
+
+                ParticularConfig.DepositKind kind = strategy.deposit_kind();
+
+                if (kind == ParticularConfig.DepositKind.SingleAsset) {
+                    asset.safeTransfer(address(strategy), allocateAmount);
+                    strategy.deposit(address(asset), allocateAmount);
+                } else if (
+                    kind == ParticularConfig.DepositKind.UniswapTwoAsset
+                ) {
+                    // since uniswap is the only one right now then the else is
+                    // okay - later in future refactor as switch/if/else like thing
+                    // does vault need to approve on strategy for transferFrom?
+                    PairReader r = PairReader(address(strategy));
+                    (address token0, address token1) = r._tokens(
+                        address(asset)
+                    );
+
+                    // figure out allocate amount right way
```

Comment:
A few questions:
 1. How do we make sure there is enough funds for token1? We know that we have at least allocateAmount of token0 in the vault but that is not guaranteed for token1
 2. We may need to adjust the way we select the strategy. See relevant discussions in #335 (I think that is more of a discussion for the long-term) and also in PR #336 (short-term fix for strategy weight).
 3. token0 and token1 could use different precision for balances. For ex I think USDT and USDC use 6 digits while DAI uses 18

Reply:
Right - this spot needs more thought - I went back and forth with tom/daniel about it too. 
1. a. we pick the lesser of the two possiblities - compute how much of other side is needed - then deposit the lesser of the two 
2. The strategy weighting needs to be more liberal - perhaps price not in %s but in terms of eth since each coin easily priceable in eth. 
3. that's fine - its deterministic scaling todo ( left as todo if others greenlight things this general approach of pr )